app/lib/features/profile/pages/my_profile_page.dart

```diff
@@ -87,26 +85,47 @@ class MyProfile extends ConsumerWidget {
     BuildContext context,
     WidgetRef ref,
   ) async {
-    final newUsername = await showDialog<String?>(
+    final TextEditingController newName = TextEditingController();
+    newName.text = profile.profile.displayName ?? '';
+
+    final newText = await showDialog<String>(
       context: context,
-      builder: (BuildContext context) => ChangeDisplayName(account: profile),
+      builder: (BuildContext context) => AlertDialog(
+        title: const Text('Change your display name'),
+        content: TextField(controller: newName),
+        actions: [
+          TextButton(
+            onPressed: () => Navigator.pop(context),
+            child: const Text('Cancel'),
+          ),
+          TextButton(
+            onPressed: () {
+              if (profile.profile.displayName != newName.text) {
+                Navigator.pop(context, newName.text);
+              } else {
+                Navigator.pop(context);
+              }
+            },
+            child: const Text('Submit'),
+          ),
+        ],
+      ),
     );
-    if (newUsername != null && context.mounted) {
+
+    if (newText != null && context.mounted) {
       showAdaptiveDialog(
```

Comment:
It becomes easy with EasyLoading ;)

Reply:
this was actually from a previous PR that is now not part of this PR anymore as it is already in `main`... If you still want to discuss that, let's address it separately.